kernel/common/array.rb

```diff
@@ -390,9 +390,25 @@ def clear
     self
   end
 
+  # Implementation based on the binomial_coefficient
+  # implemented on MRI array.c.
+  def __binomial_coefficient__(comb, size)
+    comb = size-comb if (comb > size-comb)
+    return 0 if comb < 0
+    __descending_factorial__(size, comb) / __descending_factorial__(comb, comb)
+  end
+  private :__binomial_coefficient__
```

Comment:
This should probably be moved into a mirror for the Array class (https://github.com/rubinius/rubinius/blob/master/kernel/bootstrap/array_mirror.rb), that way we don't pollute the Array class nor can Ruby code in the wild mess with it. Mirrors can be used as following: https://github.com/rubinius/rubinius/blob/master/kernel/common/array.rb#L48